src/components/AuthenticationPage.js

```diff
@@ -2,29 +2,50 @@ import Greeting from "./Greeting";
 import Header from "./Header";
 import AuthenticationForm from "./AuthenticationForm";
 
+const signup = {
+  title: "Sign Up",
+  linkMessage: "Have an account?",
+  link: "#",
+  placeholders: ["Username", "Email", "Password"],
+  name: ["username","email","password"],
+  inputType: ["text", "text", "password"],
+  buttonMessage: "Submit"
+}
+const signin = {
+  title: "Sign In",
+  linkMessage: "Need an account?",
+  link: "#",
+  placeholders: ["Email", "Password"],
+  name: ["email","password"],
+  inputType: ["text", "text"],
+  buttonMessage: "Sign in"
+}
+const userData = {}
+
 function AuthenticationPage({isSignin}){
-  const signup = {
-    title: "Sign Up",
-    linkMessage: "Have an account?",
-    link: "#",
-    placeholders: ["Username", "Email", "Password"],
-    buttonMessage: "Submit"
-  }
-  const signin = {
-    title: "Sign In",
-    linkMessage: "Need an account?",
-    link: "#",
-    placeholders: ["Email", "Password"],
-    buttonMessage: "Sign in"
-  }
   return (
     <div>
       <Header />
       <div className="flex flex-col justify-center items-center">
         <Greeting title={ isSignin ? signin.title : signup.title } linkMessage={ isSignin ? signin.linkMessage : signup.linkMessage } link={ isSignin ? signin.link : signup.link } />
-        <AuthenticationForm placeholders={ isSignin ? signin.placeholders : signup.placeholders } buttonMessage={ isSignin ? signin.buttonMessage : signup.buttonMessage } />
```

Comment:
Would be more readable if you did the check outside the authentication form. 

Have a look at this. https://reactjs.org/docs/conditional-rendering.html#inline-if-else-with-conditional-operator

Reply:
Refactored the greeting and authentication form components to conditionally render.